### ISSUE TITLE
Can't mount a volume using Blobfuse mounting path as a target directory in a Docker Container

### ISSUE THREAD

**Issue by pauloo23** [Azure/azure-storage-fuse#1530]
### Which version of blobfuse was used?
Blobfuse2

### Which OS distribution and version are you using?
Docker running on windows

### If relevant, please share your mount command.
Dockerfile:
```
    FROM mcr.microsoft.com/mirror/docker/library/ubuntu:22.04
    USER root
    RUN mkdir -p /usr/share/blobfuse2
    COPY ./config.yaml /usr/share/blobfuse2
    RUN chmod -R 777 /usr/share/blobfuse2
    
    RUN \
        mkdir -p /mnt/blobfuse_mnt && \
        mkdir -p /tmp/blobfuse_temp && \
        chmod 777 /mnt/blobfuse_mnt && \
        chmod 777 /tmp/blobfuse_temp
    
    # fix locales
    RUN apt-get update \
        && apt-get install -y --no-install-recommends locales \
        && rm -rf /var/lib/apt/lists/* \
        && localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
    
    RUN apt-get update && \
        apt-get install -y ca-certificates vim nano rsyslog wget && \
        wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb && \
        dpkg -i packages-microsoft-prod.deb && \
        apt-get update && \
        apt-get install -y libfuse3-dev fuse3 blobfuse2 && \
        rm -rf /var/lib/apt/lists/*
    
    # Create the mount script and set it to entry point once the container starts
    RUN \
        echo "/sbin/rsyslogd" > /usr/share/blobfuse2/blobfuse2-mount.sh && \
        echo "echo 'Checking if /mnt/blobfuse_mnt is mounted...'" >> /usr/share/blobfuse2/blobfuse2-mount.sh && \
        echo "if mountpoint -q /mnt/blobfuse_mnt; then echo 'Directory is mounted, unmounting now'; blobfuse2 unmount all; else echo 'Directory is not mounted'; fi" >> /usr/share/blobfuse2/blobfuse2-mount.sh && \
        echo "blobfuse2 mount all /mnt/blobfuse_mnt --config-file=/usr/share/blobfuse2/config.yaml" >> /usr/share/blobfuse2/blobfuse2-mount.sh && \
        echo "tail -f /dev/null" >> /usr/share/blobfuse2/blobfuse2-mount.sh && \
        echo "blobfuse2 unmount all" > /usr/share/blobfuse2/blobfuse2-umount.sh && \
        chmod 777 /usr/share/blobfuse2/blobfuse2-mount.sh && \
        chmod 777 /usr/share/blobfuse2/blobfuse2-umount.sh && \
        ln -s /usr/share/blobfuse2/blobfuse2-mount.sh /usr/local/bin/fuse && \
        ln -s /usr/share/blobfuse2/blobfuse2-umount.sh /usr/local/bin/unfuse && \
        ls -l /usr/local/bin && \
        ls -l /usr/share/blobfuse2
    
    
    ENTRYPOINT ["bash", "/usr/local/bin/fuse"]
```
Docker-compose.yml:
```
services:
  blobfuse2:
    build:
      context: .
      dockerfile: Dockerfile
    cap_add:
      - SYS_ADMIN
    devices:
      - /dev/fuse
    security_opt:
      - apparmor:unconfined
    stdin_open: true
    volumes:
      - blobfuse_data:/mnt/blobfuse_mnt:shared

volumes:
  blobfuse_data:

```

config.yml:
```
allow-other: true

logging:
  type: syslog
  level: log_err

components:
  - libfuse
  - file_cache
  - attr_cache
  - azstorage

libfuse:
  attribute-expiration-sec: 120
  entry-expiration-sec: 120
  negative-entry-expiration-sec: 240

file_cache:
  path: /tmp/blobfuse_temp

  timeout-sec: 120
  max-size-mb: 4096

attr_cache:
  timeout-sec: 7200

azstorage:
  type: adls
  virtual-directory: true
  use-adls: true
  account-name: *****
  account-key: *******
  endpoint: https://*****.blob.core.windows.net
  mode: key

```
### What was the issue encountered?

**Error: directory is already mounted**

### Have you found a mitigation/solution?
If I use the same files but without specifying any volume in docker-compose, it works fine!

**Comment by ashruti-msft** [comment]
Hi I tried to reproduce your case on a linux vm and got the following error :
```
Directory is mounted, unmounting now
Nothing to unmount
Error: directory is already mounted
```

The message `nothing to unmount ` is coming from the command `blobfuse2 unmount all` in your dockerfile which shows the final error `directory is already mounted` is not due to a blobfuse mount. 
Furthermore, I got the output to df /mnt/blobfuse_mnt as: 
`Filesystem     1K-blocks     Used Available Use% Mounted on`
`/dev/root       30298176 13743912  16537880  46% /mnt/blobfuse_mnt`

Looks like there is some error in your dockerfile/docker-compose files. It's not a blobfuse issue.
